coreclient/src/clients/message.rs

```diff
@@ -0,0 +1,186 @@
+// SPDX-FileCopyrightText: 2025 Phoenix R&D GmbH <hello@phnx.im>
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use anyhow::Context;
+use phnxtypes::{
+    identifiers::QualifiedUserName, messages::client_ds_out::SendMessageParamsOut, time::TimeStamp,
+};
+use rusqlite::{Connection, Transaction};
+
+use crate::{Conversation, ConversationId, ConversationMessage, MimiContent};
+
+use super::{ApiClients, CoreUser, Group, StoreNotifier};
+
+impl CoreUser {
+    /// Send a message and return it.
+    ///
+    /// Note that the message has already been sent to the DS and has internally been stored in the
+    /// conversation store.
```

Comment:
Wait, isn't the message sent to the DS in this function? Can you elaborate a bit what you mean with this doc comment?